.gitignore

```diff
@@ -1,7 +1,3 @@
-# Created by https://www.gitignore.io/api/go,linux,macos,windows,intellij,visualstudiocode
```

Comment:
A lot of this is boiler plate and environment dependant some of this should be set as a global git ignore

https://help.github.com/en/articles/ignoring-files#create-a-global-gitignore

Reply:
also I still have to delete all of idea files, we should not be including an `.idea` folder on an opensource project

Reply:
setting a global ignore means every single person has to remember to set them - there is no cost in having them here

Reply:
> we should not be including an .idea folder on an opensource project
that's more opinion than fact;  parts of the `.idea` dir can be included to help developers set-up the project faster.

I think the problem is related to the `*.iml` file.  I will experiment some more, but for now let's leave this in.

Reply:
Updated gitignore settings in #2